### DISCUSSION TITLE
Limit dependencies of the wslu package in Ubuntu

### DISCUSSION THREAD

**Discussion by vbrozik** [wslutilities/wslu#235] [Ideas]
The `wslu` package for Ubuntu (including the latest PPA for 22.04) has huge number of dependencies which are required to be installed with it:
**61 packages**, **177 MB**

The numbers were counted in the normal Ubuntu installation inside WSL2.

Currently I need just `wslview`. Installing so many packages because of that is unacceptable. So I wrote a small installer - currently just for `wslview`:
https://gist.github.com/vbrozik/2dff5d7a235484388a8e5d460d979c2e

Of course the script will break if you change the installation process of `wslview` :(

It would be great if you consider limiting the dependencies or splitting the `wslu` package so that `wslview` could be installed without them.

I think the unreasonable dependencies can prevent wider spread of the very useful tool. Maybe it is one of the reasons why it is not installed by default in Ubuntu any more.

**Comment by car-meows** [comment]
Thank you for the feedback, I will mark `imageMagick` as recommended dependency in 4.x versions so that the installation won't be that huge.

**Comment by visstro** [comment]
You can also use `--no-install-recommends`:
```
apt install wslu --no-install-recommends
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  desktop-file-utils
Suggested packages:
  imagemagick
Recommended packages:
  pulseaudio-utils x11-utils
The following NEW packages will be installed:
  desktop-file-utils wslu
0 upgraded, 2 newly installed, 0 to remove and 81 not upgraded.
Need to get 146 kB of archives.
After this operation, 480 kB of additional disk space will be used.
```

**Reply by vbrozik** [reply]
Thank you for writing the suggestion. 👍 In fact I discovered the same later after I saw that @patrick330602 suggested that he will limit the `imagemagick` dependency and I noticed that `imagemagick` was not installed by default anyway. I forgot to put this finding here. 😞

I described some of my findings as a reaction to the 4.0.0 release (I was unable to find the right commit):
https://github.com/wslutilities/wslu/discussions/240#discussioncomment-3675521
